tools/sdk-lints/src/lint_cargo_toml.rs

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use crate::anchor::replace_anchor;
+use anyhow::{bail, Context, Result};
+use cargo_toml::Manifest;
+use serde::Deserialize;
+use std::fs::read_to_string;
+use std::ops::Deref;
+use std::path::Path;
+
+#[derive(Deserialize)]
+struct DocsRsMetadata {
+    docs: Docs,
+}
+
+/// Convenience wrapper to expose fields of the inner struct on the outer struct
+impl Deref for DocsRsMetadata {
+    type Target = Metadata;
+
+    fn deref(&self) -> &Self::Target {
+        &self.docs.rs
+    }
+}
+
+#[derive(Deserialize)]
+struct Docs {
+    rs: Metadata,
+}
+
+#[derive(Deserialize)]
+struct Metadata {
+    #[serde(rename = "all-features")]
+    all_features: bool,
+    targets: Vec<String>,
+    #[serde(rename = "rustdoc-args")]
+    rustdoc_args: Vec<String>,
+}
+
+const RUST_TEAM: &str = "AWS Rust SDK Team <aws-sdk-rust@amazon.com>";
+const SERVER_CRATES: &[&str] = &["aws-smithy-http-server"];
+
+/// Check crate licensing
+///
+/// Validates that:
+/// - `[package.license]` is set to `Apache-2.0`
+/// - A `LICENSE` file is present in the same directory as the `Cargo.toml` file
+pub(crate) fn check_crate_license(path: impl AsRef<Path>) -> Result<()> {
+    let parsed = Manifest::from_path(path.as_ref()).context("failed to parse Cargo.toml")?;
+    let package = match parsed.package {
+        Some(package) => package,
+        None => bail!("missing `[package]` section"),
+    };
+    let license = match package.license {
+        Some(license) => license,
+        None => bail!("license must be specified"),
+    };
+    if license != "Apache-2.0" {
+        bail!("incorrect license: {}", license)
+    }
+    if !path
+        .as_ref()
+        .parent()
+        .expect("path must have parent")
+        .join("LICENSE")
+        .exists()
+    {
+        bail!("LICENSE file missing")
+    }
+    Ok(())
+}
+
+pub(crate) fn check_crate_author(path: impl AsRef<Path>) -> Result<()> {
+    let parsed = Manifest::from_path(path).context("failed to parse Cargo.toml")?;
+    let package = match parsed.package {
+        Some(package) => package,
+        None => bail!("missing `[package]` section"),
+    };
+    if SERVER_CRATES.contains(&package.name.as_str()) {
```

Comment:
We'd also like to enjoy this linter script :(

Reply:
everything is currently applied to both except this author check, but that's a good point, I'll add the server owner. We'll just need to keep the list of `SERVER_CRATES` up to date